### PR TITLE
consolidates to using .protocol instead of .interface and correct defines

### PR DESCRIPTION
**Pull request by jacobrosenthal**
I believe upload.interface is incorrect, and upload.protocol is the proper command
cmsisdap was mispelled
and cmsisdap will probably? always need swd option

See #111

### CODE REVIEW

**Comment by sandeepmistry** [comment]
@jacobrosenthal does this change also require a change to `tools.openocd.upload.pattern="{path}/{cmd}" {upload.verbose} -f interface/{upload.interface}.cfg -c "{upload.setup_command}" -f target/{upload.target}.cfg -c "program {{{build.path}/{build.project_name}.hex}} verify reset; shutdown;"
` line in platform.txt?

Other than that it makes sense, I forget why I decided to go for the empty value path when I started.

**Comment by jacobrosenthal** [comment]
Which means program.interface is also suspect here. 

I presume that needs to be changed to something like program.protocol, but I dont thats a real thing yet either, and seems related to https://github.com/sandeepmistry/arduino-nRF5/issues/111#issuecomment-283830118

Perhaps just leave that be until we resolve that PR upstream?

**Comment by jacobrosenthal** [comment]
So .interface is actually a totally fine custom key, however there are already a bunch of 'standards' for this with board.upload.protocol and both programmer.protocol, programmer.program.protocol 

(why both??, theyre upstream like that so Ill just leave them and duplicate them like they often do upstream https://github.com/arduino/Arduino/blob/master/hardware/arduino/avr/programmers.txt)

This PR consolidates to using .protocol instead of .interface, and attempts to populate all .protocol fields, and corrects the typo cmsisdap to cmsis-dap

**Comment by jacobrosenthal** [comment]
OK taking one more shot at clarifying this, if only for myself

Arduino defines 
https://github.com/arduino/Arduino/wiki/Arduino-IDE-1.5-3rd-party-Hardware-specification

>  •	upload a sketch to the target board (using a bootloader preinstalled on the board)
>  •	program a sketch to the target board using an external programmer
>  •	erase the target board's flash memory using an external programmer
>  •	burn a bootloader into the target board using an external programmer
> 

We don't have a boot loader, so our program and upload are identical except upload comes from board definition (defined in the core) and program comes from programmer definition (and is selected via menu by the user)

When the user hits upload we call [uploadUsingPreferences](https://github.com/arduino/Arduino/blob/master/arduino-core/src/cc/arduino/packages/uploaders/SerialUploader.java#L65)
If the user forced "using programmer" or the board defined board.upload.protocol is empty (which ours IS now as we use currently use .interface) then [uploadUsingProgrammer](https://github.com/arduino/Arduino/blob/master/arduino-core/src/cc/arduino/packages/uploaders/SerialUploader.java#L89) is actually used.
- boardPreferences go into prefs
- programmerPrefs merged ON TOP of prefs
- Then prefs is used to fill program.pattern

So currently our upload.pattern has NEVER been used. and our programmerPrefs have been wiping out our boardPreferences up til now

This changes that to fix this be aligning to upload.protocol, which WOULD skip uploadUsingProgrammer
- boardPreferences goes into prefs
- Then prefs is used to fill upload.pattern


Only tangentially related:

For uploading soft device we utilize the bootloader pathway, [uploader.burnbootloader](https://github.com/arduino/Arduino/blob/master/arduino-core/src/cc/arduino/packages/uploaders/SerialUploader.java#L352)
- boardPreferences goes into prefs
- programmerPrefs are merged ON TOP of prefs
- toolPrefs are merged ON TOP of prefs
- Then prefs is used to fill bootloader.pattern

File: platform.txt [hunk 1]
```diff
@@ -116,16 +116,16 @@ tools.openocd.cmd.windows=bin/openocd.exe
 
 tools.openocd.upload.params.verbose=-d2
 tools.openocd.upload.params.quiet=-d0
-tools.openocd.upload.pattern="{path}/{cmd}" {upload.verbose} -f interface/{upload.interface}.cfg -c "{upload.setup_command}" -f target/{upload.target}.cfg -c "program {{{build.path}/{build.project_name}.hex}} verify reset; shutdown;"
+tools.openocd.upload.pattern="{path}/{cmd}" {upload.verbose} -f interface/{upload.protool}.cfg -c "{upload.setup_command}" -f target/{upload.target}.cfg -c "program {{{build.path}/{build.project_name}.hex}} verify reset; shutdown;"
```
Comment (sandeepmistry) [inline platform.txt]: `upload.protool` - typo?

Should be `upload.protocol`?

**Comment by sandeepmistry** [comment]
Once the typo is corrected I think this is good to go.

@dlabun what do you think?

**Comment by dlabun** [comment]
My only question is how easily can this change be rolled back? Seems like a real easy way to brick the entire core by messing with the programming methods.

**Comment by jacobrosenthal** [comment]
This change just fixes our default bootloading selection, nothing more low
level than that.

And all of these devices have hardware bootloading via jlink or cmsis, so
bricking seems not to too much a worry for me.

**Comment by OwenBrotherwood** [comment]
Can this be pulled so @carlosperate can get his PR done: he is waiting for this one, so he can fix up conflicts.

**Comment by dlabun** [comment]
@jacobrosenthal @sandeepmistry Have either of you tested these changes yet?

**Comment by sandeepmistry** [comment]
Tested this with the micro:bit:

```
/Users/smistry/Library/Arduino15/packages/sandeepmistry/tools/openocd/0.10.0-dev.nrf5/bin/openocd -d2 -f interface/cmsis-dap.cfg -c transport select swd; -f target/nrf51.cfg -c program {{/var/folders/5d/pmvtw7p579n5npm6p421qf540000gn/T/arduino_build_830030/Blink.ino.hex}} verify reset; shutdown; 
```

Looks good, merging!

**Pull request merged**